### DISCUSSION TITLE
glb files not loading on fxhash sandbox

### DISCUSSION THREAD

**Discussion by joeantani** [cables-gl/cables_docs#289] [Answers & Questions] [answered]
I made a patch on cables.gl for fxhash using the fxhash op. The patch involved the randomization of a series of .glb assets. 
Everything worked fine in cables, so I exported it in zip format and downloaded it. Once once unzipped I opened the index and insterted the fxhash snipped in the head section. Then I uploaded it on the fxhash sanbox, the button was visible, the color backround too but no 3D assets was showing. After a series of attempt, I tried to use the previewer of the minter on fxhash and it turns out there everything was showing, and working properly. I assumed the sandbox doesn't allow the fetch of glb assets maybe for safety reasons(?) 
Anyone has ideas why?

**Comment by squinonescolon** [comment]
This is in their guide: https://www.fxhash.xyz/doc/artist/guide-publish-generative-token

Network requests
You cannot make any network request. If you need an external library/font/image, you must put it in your project folder. I know, this isn't an ideal workflow, but the point is to create self-contained tokens which can execute properly regardless of external conditions.

And their size limit is 30mb so you're limited on what you can do with 3d models. The best case would be to port a shader from ShaderToy of something that is similar and modify it to fit your style and subject matter and do it all in code. 

I'm working on a project and doing something that is probably similar to what you're doing, the only way I figured to do it is to use morph targets. It should work for me because I'm randomizing a single and quite simple model, but the problem is that we can't change the value property for morph targets in Cables.gl, so regrettably, I've had to use BabylonJS instead which has easy support for it.
Good luck.

**Reply by joeantani** [reply]
thank you for the precious advice.
I might read about BabylonJS.

**Reply by steam0r** [reply]
cables exports assets into the assets/ folder, and when investigating what fxhash makes out of it all the paths looked fine. this working in "preview" but not in "sandbox" sounds more of an fxhash thing to me. maybe anyone can chime in to why that is and if there is something we can do?

**Reply by squinonescolon** [reply]
Is a limitation on their part. They don't allow network calls.

**Reply by steam0r** [reply]
i read their documentation as "dont allow external network calls", just wondering why uploading assets does not work in their sandbox, but does in their preview

**Reply by squinonescolon** [reply]
My guess would be its an oversight, since as per their own rules it's not allowed.

**Reply by akerb00m** [reply]
Hi there, i am having the same problem, i have about 30 .glb files in my asset folder that doens't want to load either in the sandbox or preview. the.glb files are tiny ( the zipped folder for minting is only 858KB. just wondering, i thought no network calls means, no 'calling' something that is not Inside the zip file ? 

i asked around and one person said "cables seems to add a query to the glb URL and it might be that fxhash doesn't like it"
could tha be the problem ?
Cheers !

**Reply by steam0r** [reply]
can you maybe verify this with an image or something and loading it with `?something=test` and without?

**Reply by pandrr** [reply]
@steam0r can be tested here https://www.fxhash.xyz/sandbox

iirc this is one reason why we also want the flat export option ?

**Reply by akerb00m** [reply]
@steam0r not sure what that means "loading it with ?something=test and without"
i plan to check in tonight at the meetup, if there is time and you feel like it we could take a look then maybe ?
i talked with 'somaticbits' he said it might be more a fxhash related problem but since i am in the very early stages of learning to code i did understand it completely ; )

**Reply by steam0r** [reply]
@pandrr flat export was for objkt.com, iirc
@VoidBoid all good, will try out something on my own real quick

**Reply by akerb00m** [reply]
@steam i can quickly send over whatever, if you need something for testing, just say the word : )

**Reply by steam0r** [reply]
so, exporting and trying out this simple patch on the fxhash platform reveals that assets that have parameters attached to them do actually not get loaded:

https://dev.cables.gl/p/0BU78a

if you could/want to verify feel free to try this patch in sandbox and toggling on/off the parameters using the sidebar.

not really sure what to make of this though...is this the same once the patch is minted?

**Reply by akerb00m** [reply]
@steam0r 
i will give it a try, one thing i am thinking is that the fxhash sandbox might not accept the 'DEV test fxhash param patch' because there are random variables in there but lets see ; )

**Reply by akerb00m** [reply]
@steam0r i tested the patch in the fx sandbox, nothing happened, it did not sow the sidebar.

**Reply by steam0r** [reply]
"works" for me, did you put in the fxhash snippet as descibed on:

https://www.fxhash.xyz/doc/artist/guide-publish-generative-token

**Reply by akerb00m** [reply]
I used the fx/cables Boilerplate https://github.com/somaticbits/fxhash-cables-boilerplate
that already has the snipped included

**Reply by akerb00m** [reply]
joining the discord meetup in a sec (gertjanakerboom)

**Reply by steam0r** [reply]
that's kinda outdated so following the steps on fxhash and basically here, might work better:

https://dev.cables.gl/op/Ops.Api.FxHash.FxHash

**Reply by akerb00m** [reply]
really ? i had no idea, maybe that is the reason why it didn't work. gonna try it out !
thank you for your help !

**Reply by steam0r** [reply]
to sum up: 

- yes, fxhash does not load assets that have a get-string parameter (i.e. `?rnd=u3hdif`)
- gltfscene adds this for cachebusting, we will look into that. 

would still be nice if someone could test/verify if this is only in the sandbox (as suggested in the initial question) or also with a minted piece. (you should be able to use my patch and see if both settings of the sidebar-toggle show an image)

**Reply by akerb00m** [reply]
@steam0r aha, so the toggle button in the patch turns this (i.e. ?rnd=u3hdif) thing on and off to test it, i am happy to help where i can. I am working on getting my patch ready with the normal fxhash boilerplate, like you said. if you like i can run your testpatch through the minting process as far as possible, without actually minting it. 
also would you mind continuing this on discord, for some reason my github account got flagged and i want to know what is going on before i upload any more files etc. 
I am GertJanAkerboom#5854 at the CablesDiscord.

**Comment by steam0r** [comment, marked as answer]
putting this in a new answer as its more consice:

fxhash for some reason has a bug with/does not allow loading assets from urls that have parameters (`?some=param`) IN THEIR SANDBOX.

this leads to some ops not working properly, in this case GltfScene.

it works fine though in their preview DURING MINTING. there seems to be a difference on what they actually do there.
so if you encounter something like this in sandbox. try minting it and using that preview.

thanks to @VoidBoy for the help and directions!

**Reply by steam0r** [reply]
on dev.cables.gl GltfScene got an update and (for example) this patch now also works in the fxhash sandbox:

https://dev.cables.gl/edit/XmL8GY

**Reply by akerb00m** [reply]
awesome ! ill go and test it as soon as i have a minute. thanks !!

**Reply by akerb00m** [reply]
FYI - To anyone reading this, for the above mentioned (cables.gl / GLB) update you will need to export your project from dev.cables.gl (not cables.gl)
